### DISCUSSION TITLE
Space character resets cursor or buffer in stitch

### DISCUSSION THREAD

**Discussion by Sellular** [cc-tweaked/CC-Tweaked#1381] [General]
### Minecraft Version

1.19.x

### Version

1.101.2

### Details

Figured I'd report this since I know stitch is relatively popular in the community.

When displaying any text with a space, instead of displaying the space and proceeding text, the space is not displayed and the cursor is reset to the initial position. It then overwrites any previously written characters with the text that proceeded the space.

Reproduction code; Results in monitor_0 outputting "world!":
```
os.loadAPI( "stitch" )
stitch.setMonitors( {
    {"monitor_0", "monitor_1"},
    {"monitor_2", "monitor_3"},
} )
term.redirect( stitch )
stitch.setTextScale( 5 )
print( "Hello, world!" )
```

If `Hello, world!` is replaced with something akin to `FFFF DD` it would display `DDFF`

logs, however nothing CC related: https://gist.github.com/Sellular/7d50718ec004693ba38dfb3c25b234d0

**Comment by piprett** [comment]
This does not look like a issue with CC:T, the mod. This repo is only for CC:T, not for community creations.

**EDIT:** This message was posted on the GitHub issue, and I did not post this on the discussion, but it was automatically moved.

**Reply by Sellular** [reply]
Yeah I probably should've started it as a discussion in the first place. My bad!

**Comment by Sellular** [comment]
According to a CC discord moderator it's probably something related to text wrapping/terminal size reporting:

https://github.com/cc-tweaked/CC-Tweaked/blob/1848d9164655099b30832569e76d6e048cb1fa5d/projects/core/src/main/resources/data/computercraft/lua/bios.lua#L99-L112

However, I don't have the CC knowledge to delve into it now, nor the will to teach myself all that right now. Not looking to program outside of work right now lol. 

If anyone stumbles across this and wants to help though I'd be willing to pitch in a bit.

**Comment by SquidDev** [comment]
The problem appears to be that stitch does not increment the cursor position after a call to `blit`/`write`.

Adding the following line to the definition of `stitch.blit` (line 89) should fix this issue:

```lua
    internalX = internalX + text_len
```

**Reply by Sellular** [reply]
Dang, and that sounds like something I'd be able to figure out if I looked at the codea but longer 😂

Thank you so much for taking the time to look at this squid ❤️
